host.go

```diff
@@ -346,14 +346,22 @@ func (h *gpbftRunner) startInstanceAt(instance uint64, at time.Time) error {
 func (h *gpbftRunner) computeNextInstanceStart(cert *certs.FinalityCertificate) (_nextStart time.Time) {
 	ecDelay := time.Duration(h.manifest.EC.DelayMultiplier * float64(h.manifest.EC.Period))
 
-	ts, err := h.ec.GetTipset(h.runningCtx, cert.ECChain.Head().Key)
+	head, err := h.ec.GetHead(h.runningCtx)
 	if err != nil {
 		// this should not happen
-		log.Errorf("could not get timestamp of just finalized tipset: %+v", err)
+		log.Errorf("ec.GetHead returned error: %+v", err)
 		return h.clock.Now().Add(ecDelay)
 	}
 
-	base := ts.Timestamp()
+	// the head of the cert becomes the new base
+	baseTipSet := cert.ECChain.Head()
+	// we are not trying to fetch the new base tipset from EC as it might not be available
+	// instead we compute the relative time from the EC.Head
+	// timestamp(head) = genesis + epoch(head) * period
+	// timestamp(base) = genesis + epoch(base) * period + timestamp(head) - timestamp(head)
+	// timestamp(base) = timestamp(head) + genesis + epoch(base) * period - genesis - epoch(head) * period
+	// timestamp(base) = timestamp(head) + (epoch(base) - epoch(head)) * period
+	baseTimestamp := head.Timestamp().Add(time.Duration(baseTipSet.Epoch-head.Epoch()) * h.manifest.EC.Period)
```

Comment:
I'm going to be hypocritical here but... it's easy to have hard-to-notice off-by-one errors here. Can we extract this into a separate function and write a simple unit test?